File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/ioc/mappers/ChristmasTypes.kt
package io.micronaut.docs.ioc.mappers

import io.micronaut.core.annotation.Introspected

// tag::beans[]
@Introspected
data class ChristmasPresent (
    val packagingColor: String?,
    val type: String?,
    val weight: Float?,
    val greetingCard: String?
)

@Introspected
data class PresentPackaging (
    val weight: Float?,
    val color: String?
)

@Introspected
data class Present (
    val weight: Float?,
    val type: String?
)
// end::beans[]

File: test-suite-python/src/test/python/micronaut/docs/ioc/mappers/ChristmasTypes.py
from dataclasses import dataclass

from micronaut.core.annotation import Introspected


# tag::beans[]
@dataclass
@Introspected
class ChristmasPresent:
    packaging_color: str | None
    type: str
    weight: float
    greeting_card: str | None


@dataclass
@Introspected
class PresentPackaging:
    weight: float
    color: str


@dataclass
@Introspected
class Present:
    weight: float
    type: str
# end::beans[]
